feast-ui/rockcraft.yaml

```diff
@@ -53,30 +51,24 @@ parts:
       make build-ui
 
       # Replace the build artifacts inside the Python package
-      rm -rf /usr/local/lib/python3.10/dist-packages/feast/ui/build/
-      cp -r sdk/python/feast/ui/build /usr/local/lib/python3.10/dist-packages/feast/ui/build/
+      rm -rf /usr/local/lib/python3.12/dist-packages/feast/ui/build/
+      cp -r sdk/python/feast/ui/build /usr/local/lib/python3.12/dist-packages/feast/ui/build/
+
+      # Patch Feast CLI to explicitly use python3.12 interpreter
+      sed -i '1s|.*|#!/usr/bin/python3.12|' /usr/local/bin/feast
 
       # Promote Python packages and CLI binaries to the rock install directory
-      mkdir -p $CRAFT_PART_INSTALL/usr/local/lib/python3.10/dist-packages
+      mkdir -p $CRAFT_PART_INSTALL/usr/local/lib/python3.12/dist-packages
       mkdir -p $CRAFT_PART_INSTALL/usr/local/bin
-      cp -fr /usr/local/lib/python3.10/dist-packages/* $CRAFT_PART_INSTALL/usr/local/lib/python3.10/dist-packages/
+      cp -fr /usr/local/lib/python3.12/dist-packages/* $CRAFT_PART_INSTALL/usr/local/lib/python3.12/dist-packages/
       cp -fr /usr/local/bin/* $CRAFT_PART_INSTALL/usr/local/bin/
 
-
   non-root-user:
     plugin: nil
     after: [feast-ui]
-    overlay-script: |
-      # Create non-root user 'ubuntu' with fixed UID/GID inside the overlay
-      groupadd -R $CRAFT_OVERLAY -g 1001 ubuntu
-      useradd -R $CRAFT_OVERLAY -M -r -u 1001 -g ubuntu ubuntu
     override-prime: |
       craftctl default
 
-      # Create and assign home directory
-      mkdir -p $CRAFT_PRIME/home/ubuntu
-      chown -R 1001:1001 $CRAFT_PRIME/home/ubuntu
-
       # Ensure Feast UI build directory is writable for runtime metadata
-      mkdir -p $CRAFT_PRIME/usr/local/lib/python3.10/dist-packages/feast/ui/build
-      chown -R 1001:1001 $CRAFT_PRIME/usr/local/lib/python3.10/dist-packages/feast/ui/build
+      mkdir -p $CRAFT_PRIME/usr/local/lib/python3.12/dist-packages/feast/ui/build
+      chown -R 584792:584792 $CRAFT_PRIME/usr/local/lib/python3.12/dist-packages/feast/ui/build
```

Comment:
suggestion(non-blocking): add a comment saying this is giving permissions to `_daemon_` user with reference to rockcraft docs on the UID: https://documentation.ubuntu.com/rockcraft/en/latest/reference/rockcraft.yaml/#run-user